versioned_docs/version-V3/guides/governance/license-modifications.md

```diff
@@ -0,0 +1,171 @@
+
+```typescript
+import { ethers } from "hardhat";
+import { waffle } from "hardhat";
+import { Contract } from "ethers";
+import {
+  GOVERNOR_BRAVO_ABI,
+  ENS_REGISTRY_ABI,
+  ENS_PUBLIC_RESOLVER_ABI,
+} from "./utils";
+import { namehash } from "@ethersproject/hash";
+import { keccak256 } from "@ethersproject/keccak256";
+import { utils } from "ethers";
+import { Interface } from "@ethersproject/abi";
+import "hardhat";
+
+const { provider } = waffle;
+
+// get the governor bravo contract
+const governorBravoAddress = "0x408ED6354d4973f66138C91495F2f2FCbd8724C3";
+const governorBravo = new Contract(
+  governorBravoAddress,
+  GOVERNOR_BRAVO_ABI,
+  provider
+);
+
+const NODE_TOP_LEVEL: string = namehash("uniswap.eth");
+const LABEL: string = keccak256(utils.toUtf8Bytes("v3-core-license-grants"));
+const OWNER_UNISWAP_GOVERNANCE_TIMELOCK: string =
+  "0x1a9C8182C09F50C8318d769245beA52c32BE35BC";
+const RESOLVER_PUBLIC_ENS_RESOLVER: string =
+  "0x4976fb03c32e5b8cfe2b6ccb31c09ba78ebaba41";
+const TTL: number = 0;
+
+const ensRegistryInterface = new Interface(ENS_REGISTRY_ABI);
+const setSubnodeRecordCalldata = ensRegistryInterface.encodeFunctionData(
+  "setSubnodeRecord",
+  [
+    NODE_TOP_LEVEL,
+    LABEL,
+    OWNER_UNISWAP_GOVERNANCE_TIMELOCK,
+    RESOLVER_PUBLIC_ENS_RESOLVER,
+    TTL,
+  ]
+);
+
+const ensPublicResolverInterface = new Interface(ENS_PUBLIC_RESOLVER_ABI);
+
+const NODE: string = namehash("v3-core-license-grants.uniswap.eth");
+const KEY: string = "[your-project's-additional-use-grant-title]";
+const VALUE: string = "[your-additional-use-grant-description]";
+const setTextCalldata = ensPublicResolverInterface.encodeFunctionData(
+  "setText",
+  [NODE, KEY, VALUE]
+);
+
+const ENS_REGISTRY_ADDRESS: string =
+  "0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e";
+
+const PUBLIC_ENS_RESOLVER_ADDRESS: string =
+  "0x4976fb03c32e5b8cfe2b6ccb31c09ba78ebaba41";
+
+/// @notice the ordered list of target addresses for calls to be made
+const targets = [ENS_REGISTRY_ADDRESS, PUBLIC_ENS_RESOLVER_ADDRESS];
+
+/// @notice The ordered list of values (i.e. msg.value) to be passed to the calls to be made.
+/// as this example does not include the transfering of any tokens, this list is empty.
+const values = [0, 0];
+
+/// @notice The ordered list of function signatures to be called. The signatures arguments
+/// are optional, if not provided, the function signature will be inferred from the calldata
+const signatures = ["", ""];
+
+/// @notice The ordered list of calldata to be passed to each call in the proposal. The calldata
+/// in this example takes the place of the function signature arguments.
+const calldatas = [setSubnodeRecordCalldata, setTextCalldata];
+
+/// @notice the description of the proposal.
+const description = "Your Additional Use Grant Proposal Description";
+
+async function main() {
+  const yourSigner = await ethers.getSigner("YOUR_SIGNER_ADDRESS_HERE");
+
+  return Promise.all(
```

Comment:
just need to await the promise, and you don't need the .all - also might want to log the tx hash or something